packages/core/src/App/Components/Layout/Header/wallets/account-info-wallets.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { CSSTransition } from 'react-transition-group';
+import { Icon, WalletIcon, Text, AppLinkedWithWalletIcon } from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import { formatMoney, getCurrencyDisplayCode } from '@deriv/shared';
+import { useStore, observer } from '@deriv/stores';
+import { getActiveLoginIDFromLocalStorage } from '@deriv/utils';
+import { useStoreWalletAccountsList, useStoreLinkedWalletsAccounts } from '@deriv/hooks';
+import { AccountSwitcherWallet, AccountSwitcherWalletMobile } from 'App/Containers/AccountSwitcherWallet';
+import { AccountsInfoLoader } from '../Components/Preloader';
+import AccountInfoWrapper from '../account-info-wrapper';
+import WalletBadge from './wallet-badge';
+
+type TAccountInfoWallets = {
+    toggleDialog: () => void;
+    is_dialog_on: boolean;
+};
+
+type TDropdownArrow = {
+    is_disabled?: boolean;
+};
+
+type TBalanceLabel = {
+    balance: number;
+    currency: string;
+    is_virtual: boolean;
+    display_code: string;
+};
+
+type TMobileInfoIcon = {
+    wallet_account: ReturnType<typeof useStoreWalletAccountsList>['data'][number];
+};
+
+type TDesktopInfoIcons = {
+    wallet_account: ReturnType<typeof useStoreWalletAccountsList>['data'][number];
+};
+
+const DropdownArrow = ({ is_disabled = false }: TDropdownArrow) =>
+    is_disabled ? (
+        <Icon data_testid='dt_lock_icon' icon='IcLock' />
+    ) : (
+        <Icon data_testid='dt_select_arrow' icon='IcChevronDownBold' className='acc-info__select-arrow' />
+    );
+
+const BalanceLabel = ({ balance, currency, is_virtual, display_code }: Partial<TBalanceLabel>) =>
+    typeof balance !== 'undefined' || !currency ? (
+        <div className='acc-info__wallets-account-type-and-balance'>
+            <Text
+                as='p'
+                data-testid='dt_balance'
+                className={classNames('acc-info__balance acc-info__wallets-balance', {
+                    'acc-info__balance--no-currency': !currency && !is_virtual,
+                })}
+            >
+                {!currency ? (
+                    <Localize i18n_default_text='No currency assigned' />
+                ) : (
+                    `${formatMoney(currency, balance ?? 0, true)} ${display_code}`
+                )}
+            </Text>
+        </div>
+    ) : null;
+
+const MobileInfoIcon = observer(({ wallet_account }: TMobileInfoIcon) => {
```

Comment:
Can we just pass whats necessary instead of the whole reponse

Reply:
Updated.